### PR TITLE
Make shadowing attributes a warning instead of an error

### PR DESCRIPTION
**Pull request by adriangb**
Closes https://github.com/pydantic/pydantic/issues/7108

Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by adriangb** [comment]
please review

File: tests/test_main.py [hunk 1]
```diff
@@ -2887,3 +2887,38 @@ class Model(BaseModel):
             'ctx': {'max_length': 1},
         }
     ]
+
+
+def test_shadow_attribute() -> None:
+    """_summary_"""
```
Comment (Kludex) [inline tests/test_main.py:2893]: 👀

Comment (adriangb) [reply tests/test_main.py:2893]:
```suggestion
    """https://github.com/pydantic/pydantic/issues/7108"""
```

**Review by samuelcolvin — APPROVED** [review]
LGTM

**Pull request merged**